index.ts

```diff
@@ -26,7 +26,23 @@ export function getManifestPermissionsSync(): Required<chrome.permissions.Permis
 	return manifestPermissions;
 }
 
-export async function getAdditionalPermissions(): Promise<Required<chrome.permissions.Permissions>> {
+interface Options {
+	strictOrigins?: boolean;
+}
+
+const hostRegex = /:[/][/]([^/]+)/;
+function parseDomain(origin: string): string {
+	return origin
+	 	// Extract host
+		.split(hostRegex)[1]!
+
+		// Discard anything but the first- and second-level domains
+		.split('.')
+		.slice(-2)
+		.join('.');
+}
```

Comment:
This is an extremely ugly way to extract `google.com` from `*://*.google.com/*`. Is there a non-awful regex that reliably picks _only_ the `A.B` part?